### ISSUE TITLE
what batch size are we actually using when we are using 8 gpus 

### ISSUE THREAD

**Issue by 652994331** [NVIDIA/DeepLearningExamples#409]
Hi, a quick question, if we are using the code with report_loss = FALSE. and 8 gpus. what batch size are we actually using? train_batch_size you input or train_batch_size * 8 ?
if it's latter, could you please show me where the  scale calculation is in the code?
(I only found it in if report_loss = true, that's why i ask if we use report_loss = False what gonna happen then). Thanks

**Comment by swethmandava** [comment]
Covered in #368

**Comment by 652994331** [comment]
@swethmandava  Thanks for your reply, really appreciate. However, the part I cant understand is, as the screenshot below, it seems only if we use report _loss, the training_hook would add global_batch size or it's just normal batch.(like line 512 and 517 ).
![image](https://user-images.githubusercontent.com/51428350/75723639-87686300-5c91-11ea-9bda-47dab4a50bbe.png)

**Comment by swethmandava** [comment]
It is a logging hook - only used to print data. Have a look [here](https://github.com/NVIDIA/DeepLearningExamples/blob/8674bb6a6cd439c3a08654e828160e5a927ee6c7/TensorFlow/LanguageModeling/BERT/run_pretraining.py#L120) at what it does. If report_loss=False, I do not need to print data and hence do not need that hook.

**Comment by 652994331** [comment]
@swethmandava  ok, this is a logging hook. When i was look at line 533, it's said: batch size = FLAGS.train_batch_size. is this ok? i  mean can you show me where the real batch size is calculated, is this because the data's shard? I am sorry first time to distribute training, kinda overwhelming

**Comment by swethmandava** [comment]
It is not explicitly calculated, it is implicit. 

My data loader is loading one batch of data with batch size = train_batch_size for every iteration. but this is done on ALL workers. hence, implicitly, the global_batch_size is train_batch_size * num_workers. 

Helpful link: 
[Distributed Deep Learning with Horovod](https://developer.download.nvidia.com/video/gputechconf/gtc/2019/presentation/s9321-distributed-deep-learning-with-horovod.pdf)

**Comment by 652994331** [comment]
@swethmandava  ok, so finally i understand, the  key is the data loader part, according to my understanding of your words, every ieration, different workers load different data shard  and this is done by shuffle data shards, right?
